src/components/sbb-radio-button-group/sbb-radio-button-group.tsx

```diff
@@ -0,0 +1,182 @@
+  })
+  public didChange: EventEmitter<any>;
+
+  public connectedCallback(): void {
+    this._updateRadios();
+    this._namedSlots = queryAndObserveNamedSlotState(this._element, this._namedSlots);
+  }
+
+  @Listen('sbbNamedSlotChange', { passive: true })
+  public handleSlotNameChange(event: CustomEvent<Set<string>>): void {
+    this._namedSlots = queryNamedSlotState(this._element, this._namedSlots, event.detail);
+  }
+
+  @Listen('sbb-radio-button_did-select', { passive: true })
+  public onRadioButtonSelect(event: CustomEvent<Set<string>>): void {
+    this.value = event.detail;
+  }
+
+  private _updateRadios(): void {
+    const radios = this._radioButtons;
+    const value = this.value ?? radios.find((radio) => radio.checked)?.value;
+
+    for (const radio of radios) {
+      radio.name = this.name;
+      radio.checked = radio.value === value;
+      radio.disabled = radio.disabled ? radio.disabled : this.disabled;
+      radio.required = radio.required ? radio.required : this.required;
+      radio.tabIndex = radio.checked && !radio.disabled ? 0 : -1;
+      radio.size = this.size;
+      radio.allowEmptySelection = this.allowEmptySelection;
+    }
+
+    // If no option is selected, make the first one focusable
+    radios.length && (radios[0].tabIndex = value || radios[0].disabled ? radios[0].tabIndex : 0);
+  }
+
+  private get _radioButtons(): InterfaceSbbRadioButton[] {
+    return Array.from(
+      this._element.querySelectorAll('sbb-radio-button')
+    ) as InterfaceSbbRadioButton[];
+  }
+
+  private _getEnabledRadios(): InterfaceSbbRadioButton[] {
+    return this._radioButtons.filter((r) => !r.hasAttribute('disabled'));
+  }
+
+  @Listen('keydown')
+  public handleKeyDown(evt: KeyboardEvent): void {
+    const enabledRadios = this._getEnabledRadios();
+    const checked = enabledRadios.findIndex((t) => t.checked);
+    const cur = checked !== -1 ? checked : 0;
+    const size = enabledRadios.length;
+    const prev = cur === 0 ? size - 1 : cur - 1;
+    const next = cur === size - 1 ? 0 : cur + 1;
+
+    // don't trap nested handling
+    if (
+      (evt.target as HTMLElement) !== this._element &&
+      (evt.target as HTMLElement).parentElement !== this._element
+    ) {
+      return;
+    }
+
+    const currentWritingMode = getDocumentWritingMode();
+    const prevKey = currentWritingMode === 'rtl' ? 'ArrowRight' : 'ArrowLeft';
+    const nextKey = currentWritingMode === 'rtl' ? 'ArrowLeft' : 'ArrowRight';
+
+    if (evt.key === prevKey || evt.key === 'ArrowUp') {
+      enabledRadios[prev].select(this.allowEmptySelection);
+      enabledRadios[prev].focus();
+      evt.preventDefault();
+    } else if (evt.key === nextKey || evt.key === 'ArrowDown') {
+      enabledRadios[next].select(this.allowEmptySelection);
+      enabledRadios[next].focus();
+      evt.preventDefault();
+    }
+  }
+
+  public render(): JSX.Element {
+    return (
+      <Host role="radiogroup" aria-label={this.name}>
```

Comment:
Since we do not have a label in the radio-group we use the `aria-label` attribute. As reported in the ARIA: radio role MDN page, "The containing radiogroup element should either have a visible label referenced by aria-labelledby or a label specified with aria-label."

Reply:
That should be the autogenerated one. According to the spec: "name should be optional and default to an autogenerated one." So normally, the user should provide a more explanatory label otherwise the default one is used.